qemu/tests/thin_provisioning_optimize.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+import re
+
+from autotest.client.shared import error
+from avocado.core import exceptions
+from avocado.utils import process
+from virttest import utils_misc
+from virttest import data_dir
+
+
+@error.context_aware
+def run(test, params, env):
+
+    standard_time = 0.4
+    realtime = None
+    qemu_img_binary = utils_misc.get_qemu_img_binary(params)
+    base_dir = params.get("images_base_dir", data_dir.get_data_dir())
+    if not qemu_img_binary:
+        raise exceptions.TestError("Can't find the command qemu-img.")
+
+    image_create_cmd = params.get("create_cmd")
+    image_create_cmd = image_create_cmd % (qemu_img_binary, base_dir)
+    image_convert_cmd = params.get("convert_cmd")
+    image_convert_cmd = image_convert_cmd % (qemu_img_binary, base_dir, base_dir)
+    process.system_output(image_create_cmd, shell=True)
+    output = process.system_output(image_convert_cmd, shell=True)
+    realtime = re.search(r"real\s+\dm(.*)s", output)
+    if realtime is None:
+        raise exceptions.TestError("Faild to get the realtime")
+    realtime = float(realtime.group(1))
+    logging.info("real time is : {:f}".format(realtime))
+    if realtime >= standard_time:
+        err = "realtime({:f}) to convert the image is a little longer than standardtime({:f})"
+        raise exceptions.TestFail(err.format(realtime, standard_time))
+
+    delete_image = params.get("disk_name")
+    delete_image = os.path.join(base_dir, delete_image)
+    delete_convert_image = params.get("convert_disk_name")
+    delete_convert_image = os.path.join(base_dir, delete_convert_image)
+    if not delete_image or not delete_convert_image:
```

Comment:
view from you code, not fond anything related to TP, can you share me the case link via xchat?